WordPress/src/main/java/org/wordpress/android/ui/posts/editor/ImageEditorInitializer.kt

```diff
@@ -70,5 +88,70 @@ class ImageEditorInitializer {
             } else {
                 throw(IllegalArgumentException(IMAGE_STRING_URL_MSG))
             }
+
+        private fun onEditorAction(): (EditorAction) -> Unit = { action ->
+            trackEditorAction(action)
+        }
+
+        private fun trackEditorAction(action: EditorAction) {
+            val stat = when (action) {
+                is EditorShown -> Stat.MEDIA_EDITOR_SHOWN
+                is EditorFinishedEditing -> Stat.MEDIA_EDITOR_USED
+                is EditorCancelled,
+                is PreviewImageSelected,
+                is PreviewInsertImagesClicked,
+                is PreviewCropMenuClicked,
+                is CropOpened,
+                is CropDoneMenuClicked,
+                is CropSuccessful -> null
+            }
+
+            val properties = when (action) {
+                is EditorShown -> mapOf(NUMBER_OF_IMAGES to action.numOfImages)
+                is EditorFinishedEditing -> {
+                    val distinctActions = action.actions.distinct()
+                    if (distinctActions.isNotEmpty()) {
+                        mapOf(ACTIONS to distinctActions.map { it.label })
+                    } else {
+                        null
+                    }
+                }
+                is EditorCancelled,
+                is PreviewImageSelected,
+                is PreviewInsertImagesClicked,
+                is PreviewCropMenuClicked,
+                is CropOpened,
+                is CropDoneMenuClicked,
+                is CropSuccessful -> null
+            }
+
+            val noEditActionPerformed = stat == Stat.MEDIA_EDITOR_USED && properties == null
+            if (stat == null || noEditActionPerformed) {
+                return
+            }
+
+            if (properties == null) {
+                AnalyticsTracker.track(stat)
+            } else {
+                AnalyticsTracker.track(stat, properties)
+            }
+
+            // Also track "media_editor" source with media properties
+            if (action is EditorFinishedEditing) {
+                trackAddMediaFromMediaEditor(action.outputDataList)
+            }
+        }
+
+        private fun trackAddMediaFromMediaEditor(outputDataList: List<OutputData>) {
```

Comment:
Do we want to track this event even when someone uses the ImageEditor in a different flow - such as "Choose Avatar" in the "Me" section? I'd consider tracking this event within the EditPostActivity as it seems unrelated to the ImageEdtior itself. Wdyt?

Reply:
Very futuristic of you! Will do as suggested 😀.

Reply:
Tbh I usually don't plan for something that might happen but probably never will. But in this case it felt more like fixing the SRP than planning for future. ;)